A visitor for exceeding canvas (and margin) by a complex command

Dutchu opened the pull request:
CommandVisitor pattern redesign for stateless functionality with Java's Generics. Implemented CommandBoundaryVisitor for checks of Canvas bounds before running a Command. Redesigned other CommandVisitor implementations to match Generic and stateless Visitor design.

Dutchu committed:
A visitor for exceeding canvas (and margin) by a complex command

CommandVisitor pattern redesign for stateless with additional Functionality to be Generic. Implemented CommandBoundaryVisitor for checks of Canvas bounds before running a Command. Redesigned other CommandVisitor implementations to match Generic and stateless Visitor design.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java:
```diff
@@ -14,10 +15,11 @@
     private DriverCommand currentCommand = null;
     private Publisher changePublisher = new Publisher();
-    private List<CommandVisitor> visitorList= new ArrayList<CommandVisitor>();
+    private final CommandVisitor<Map<String, Integer>> countingVisitor = new CommandCounterVisitor();
+    private final CommandVisitor<Boolean> boundaryVisitor = new CommandBoundaryVisitor();
```
Previous soultion was more generic. I would prefer avoiding to depend on more particular visitors in DriverCommandManager class. 
I suggest following observer pattern by moving both visitors to separate Subscriber which would be responsible to calculate and update "visitor message" in CommandManagerWindow. It would be more SRP.

Dutchu replied:
[Observer pattern for Command Visitor]
(https://github.com/iis-powp-2024z/powp_jobs2d_projects/pull/28/commits/85f63f632ea44a3febd82be7706952fc8c455631)
Created default observers for Visitor's Logging capabilities and a List for additional Logging ideas.

kowallus reviewed (changes requested):
One more thing to approve.

Dutchu committed:
Observer pattern for Command Visitor

Created default observers for Visitor's Logging capabilities and a List for additional Logging ideas.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java:
```diff
@@ -15,18 +18,30 @@
+    private final List<CommandVisitorSubscriber> visitorSubscribers = new ArrayList<>();
+    // Constructor to add default subscribers
+    public DriverCommandManager() {
+        visitorSubscribers.add(new BoundaryVisitorSubscriber());
+        visitorSubscribers.add(new CounterVisitorSubscriber());
```
Boundary and Counter visitors should be added externally (to avoid dependencies)

kowallus reviewed (approved):
OK. However, dependencies should be handled better.
